worker/src/main/java/org/apache/celeborn/service/deploy/worker/congestcontrol/TimeSlidingHub.java

```diff
@@ -105,11 +105,15 @@ public synchronized void add(long currentTimestamp, N newNode) {
       }
 
       _deque.add(Pair.of(lastNode.getLeft() + intervalPerBucketInMills, (N) newNode.clone()));
-      sumNode.combineNode(newNode);
+      N nodeToCombine = sumInfo.getLeft();
+      nodeToCombine.combineNode(newNode);
+      sumInfo = Pair.of(nodeToCombine, sumInfo.getRight() + (int) nodesToAdd);
```

Comment:
`(int) nodesToAdd` if the cast is safe, let's do cast at the variable definition, otherwise use long instead

Reply:
DOne